### DISCUSSION TITLE
Plots used to work but now don't.  Also, the error message doesn't make sense,

### DISCUSSION THREAD

**Discussion by pnachtwey** [wxMaxima-developers/wxmaxima#1814] [General]
I have a problem graphing position, speed and acceleration.  This program has worked for many years until I installed the new version of wxMaxima

Compute the position, velocity and acceleration for all the motion segments
````
 -->	pva(t):= (
	    if t<t01 then
	        ( t: t-t0, [x0+v0*t+(a0/2)*t^2+(j/6)*t^3, v0+a0*t+(j/2)*t^2,a0+j*t] )
	    elseif t1<=t and t<t2 then
	        ( t: t-t1, [ x1+v1*t+(a/2)*t^2,v1+a*t,a ] )
	    elseif t2<=t and t<t3 then
	        ( t: t-t2, [x2+v2*t+(a/2)*t^2-(j/6)*t^3, v2+a*t-(j/2)*t^2, a-j*t] )
	    elseif t3<=t and t<t4 then
	        ( t: t-t3, [x3+v*t, v, 0 ] )
	    elseif t4<=t and t<t5 then
	        ( t: t-t4, [ x4+v*t-(j/6)*t^3, v-(j/2)*t^2, -j*t] )
	    elseif t5<=t and t<t6 then
	        ( t: t-t5, [ x5+v5*t-(d/2)*t^2, v5-d*t, -d ] )
	    elseif t6<=t and t<t7 then
	        ( t: t-t6, [ x6+v6*t-(d/2)*t^2+(j/6)*t^3,v6-d*t+(j/2)*t^2, -d+j*t ] )
	    else
	        ( t: t-t7, [ x7+v7*t+(a7/2)*t^2 ,v7+a7*t,a7 ] )
	)$
````

Plot the position, velocity and acceleration for all the motion segments
````
 -->	wxplot2d(pva(t)[1],[t,t0,t8],[xlabel,"Time"],[ylabel,"Position"],[legend,"Postion"]);
	wxplot2d(pva(t)[2],[t,t0,t8],[xlabel,"Time"],[ylabel,"Velocity"],[legend,"Velocity"]);
	wxplot2d(pva(t)[3],[t,t0,t8],[xlabel,"Time"],[ylabel,"Acceleration"],[legend,"Acceleration"]);

Maxima encountered a Lisp error:

 The value

   "Postion"

 is not of type

   LIST

Automatically continuing.

To enable the Lisp debugger set *debugger-hook* to nil.
````

If I remove the `[legend,"Position"]` then it works.   The "Position" is a string, not a list.  This applies to the velocity and acceleration graphs too

**Comment by gunterkoenigsmann** [comment]
(Edit: Markdown that makes drag-and-drop into wxMaxima easier).

I get a different error message as I don't know the values for t0, t8 etc. ...but I found a minimal example for reproducing that error:

````
plot2d(sin(t),[t,0,8],[legend,"Postion"]);
````
Will ask the maxima-discuss list if they know what is amiss.

**Reply by gunterkoenigsmann** [reply]
They found the bug, but fixing it has introduced another bug. Seems like the real solution will need one or two days of time.

**Reply by pnachtwey** [reply]
No hurry.  I was just updating my programs.  This is an old file
https://deltamotion.com/peter/wxMaxima/Seg1234567.wxmx
You can see it did work long ago.  Now you have the numbers.

**Comment by gunterkoenigsmann** [comment]
I believe that the problem has now been solved by the maxima team. Thanks a lot for reporting it. Normally I would now ask you if you are using MS Windows and, if yes, would tell you that at https://wolfgang.dautermann.at/maxima/nightlybuild/ you'll find an installer with all the problems fixed. But currently I am not sure if "copy as bitmap" works in that installer so your mileage with that installer might vary.

**Comment by pnachtwey** [comment]
I have Windows. I will test it now.
I tested wxmaxima.exe in maxima-current on Windows 11.

oops,  I got this error
(%i1)
wxplot2d(pva(t)[1],[t,t0,t8],[xlabel,"Time"],[ylabel,"Position"],[legend,"Position"]);
wxplot2d(pva(t)[2],[t,t0,t8],[xlabel,"Time"],[ylabel,"Velocity"],[legend,"Velocity"]);
wxplot2d(pva(t)[3],[t,t0,t8],[xlabel,"Time"],[ylabel,"Acceleration"],[legend,"Acceleration"])
;
plotting: range must be of the form [variable, min, max]; found: [t,t0,t8]

 -- an error. To debug this try: debugmode(true);

I have the range correct but I got an error

BTW, I have an Intel Mac Book Pro.  I run wxMaxima using brew.
I have a Raspberry Pi.  It runs wxMaxima.
I have a Linux Mint Debian Edition.

**Comment by gunterkoenigsmann** [comment]
That error message indicates that Maxima believes that `t0` and `t8` haven't been assigned a value to before trying to plot. Might that somehow be the case?
